src/redux/books/booksThunks.js

```diff
@@ -0,0 +1,58 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/HYYvKmGO3upxaXekhp4l/books';
+const delURL = (id) => `${URL}/${id}`;
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await fetch(URL)
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res.json();
+    })
+    .catch((err) => err.message);
+  return response;
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  const response = await fetch(URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  })
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res;
+    })
+    .catch((err) => err.message);
+  if (response.statusText !== 'Created') {
+    return 'failed';
+  }
+  return book;
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  const response = await fetch(delURL(id), {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ item_id: id }),
+  })
+    .then((res) => {
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return res;
+    })
+    .catch((err) => err.message);
+  if (response.statusText !== 'Created') {
+    return 'failed';
+  }
+  return id;
+});
```

Comment:
- [ ] There is an issue in the add and remove functions that take action after refreshing the page, please 🙏🏽 correct this behavior for a better UX.

>  When a user adds or removes a book, the change is only visible after the user refreshes the page. Kindly consider making the change visible immediately after a book has been added or removed by updating the redux store after the book has been successfully added/removed.